### ISSUE TITLE
In setdata i add data-attr in json but that not show while form rendering can i add attribute in json??

### ISSUE THREAD

**Issue by 1995ankita** [kevinchappell/formBuilder#1495]
### Environment Details:

 - formBuilder Version: 
 - Browser: 
 - OS:

**Comment by lucasnetau** [comment]
Yes, data attributes are supported. 

Please complete the issue template with required information and your replication code for us to assist you further.

**Comment by 1995ankita** [comment]
<script src="{{ URL::asset('assets/js/jquery-3.7.1.min.js') }}"></script>
<script src="{{ URL::asset('assets/bootstrap-5.3.2-dist/js/bootstrap.min.js') }}"></script>
<div class="card">
    <div class="card-body">
        <label for="name">Name</label>
        <input type="text" id="name" name="name" class="form-control" />
        <div id="fb-editor"></div>
        <pre><code id="markup"></code></pre>
    </div>
</div>

<script src="https://cdnjs.cloudflare.com/ajax/libs/jqueryui/1.13.2/jquery-ui.min.js"></script>
<script src="{{ URL::asset('assets/form-builder/form-builder.min.js') }}"></script>
<script>
    var fbEditor = document.getElementById('fb-editor');



    var formBuilder = $(fbEditor).formBuilder({
        onSave: function(evt, formData) {
            console.log(formData);
            saveForm(formData);
        `},`
    });


    $(function() {
        $.ajax({
            type: 'get',
            headers: {
                'Authorization': 'Bearer ' + localStorage.getItem('token')
            },
            url: '{{ URL('get-form-builder-edit') }}',
            data: {
                'id': '{{ $id }}'
            },
            success: function(data) {
                console.log(data.content.data_id);
                $("#name").val(data.form.name);
               formBuilder.actions.setData(data.content);
            }
        });
    });

    function saveForm(form) {
        $.ajax({
            type: 'post',
            headers: {
                'Authorization': 'Bearer ' + localStorage.getItem('token')
            },
            url: '{{ URL('update-form-builder') }}',
            data: {
                'form': form,
                'name': $("#name").val(),
                'id': {{ $id }},
                "_token": "{{ csrf_token() }}",
            },
            success: function(data) {
                //location.href = "/form-builder";
            }
        });
    }
</script>

**Comment by 1995ankita** [comment]
```<script src="{{ URL::asset('assets/js/jquery-3.7.1.min.js') }}"></script>
<script src="{{ URL::asset('assets/bootstrap-5.3.2-dist/js/bootstrap.min.js') }}"></script>
<div class="card">
    <div class="card-body">
        <label for="name">Name</label>
        <input type="text" id="name" name="name" class="form-control" />
        <div id="fb-editor"></div>
        <pre><code id="markup"></code></pre>
    </div>
</div>

<script src="https://cdnjs.cloudflare.com/ajax/libs/jqueryui/1.13.2/jquery-ui.min.js"></script>
<script src="{{ URL::asset('assets/form-builder/form-builder.min.js') }}"></script>
<script>
    var fbEditor = document.getElementById('fb-editor');
    var formBuilder = $(fbEditor).formBuilder({
        onSave: function(evt, formData) {
            console.log(formData);
            saveForm(formData);
        },
    });


    $(function() {
        $.ajax({
            type: 'get',
            headers: {
                'Authorization': 'Bearer ' + localStorage.getItem('token')
            },
            url: '{{ URL('get-form-builder-edit') }}',
            data: {
                'id': '{{ $id }}'
            },
            success: function(data) {
                console.log(data.content.data-id);
                $("#name").val(data.form.name);
                formBuilder.actions.setData(data.content);
            }
        });
    });```
      Here i m sending data-id attribute in  ajax call that i want to set in formbuilder while setdata after that i want to get that data-id while saving formbuilder.

**Comment by 1995ankita** [comment]
Here i m sending data-id attribute in  ajax call that i want to set in formbuilder while setdata after that i want to get that data-id while saving formbuilder.
    
  ```<script src="{{ URL::asset('assets/js/jquery-3.7.1.min.js') }}"></script>
<script src="{{ URL::asset('assets/bootstrap-5.3.2-dist/js/bootstrap.min.js') }}"></script>
<div class="card">
    <div class="card-body">
        <label for="name">Name</label>
        <input type="text" id="name" name="name" class="form-control" />
        <div id="fb-editor"></div>
        <pre><code id="markup"></code></pre>
    </div>
</div>

<script src="https://cdnjs.cloudflare.com/ajax/libs/jqueryui/1.13.2/jquery-ui.min.js"></script>
<script src="{{ URL::asset('assets/form-builder/form-builder.min.js') }}"></script>
<script>
    var fbEditor = document.getElementById('fb-editor');
    var formBuilder = $(fbEditor).formBuilder({
        onSave: function(evt, formData) {
            console.log(formData);
            saveForm(formData);
        },
    });


    $(function() {
        $.ajax({
            type: 'get',
            headers: {
                'Authorization': 'Bearer ' + localStorage.getItem('token')
            },
            url: '{{ URL('get-form-builder-edit') }}',
            data: {
                'id': '{{ $id }}'
            },
            success: function(data) {
                console.log(data.content.data-id);
                $("#name").val(data.form.name);
                formBuilder.actions.setData(data.content);
            }
        });
    });```

**Comment by lucasnetau** [comment]
Looking at your code the data-id is not part of formBuilder at all, but an attribute you are setting from a template and using it to load the form from a database. Save the ID to a const further up in your code and reference this in both saveForm and data: in ajax.

Your other option is to set an data-* attribute onto fbEditor and use that where required.

**Comment by 1995ankita** [comment]
@lucasnetau  i m trying to implement data-id for each field in formbuilder.

**Comment by lucasnetau** [comment]
Kevin provided you a codepen solution to this question https://github.com/kevinchappell/formBuilder/issues/1498, you need to define the data attribute in the formData before you can assign a value
